tests/store/store_utils_tests.cpp

```diff
@@ -386,48 +412,39 @@ void vencode_from_array(T expected_value, size_t expected_length) {
 
 #ifdef IRESEARCH_SSE2
 
-void read_write_optimized(const std::vector<uint32_t>& source, std::vector<uint32_t>& enc_dec_buf) {
-  // write block
-  irs::bstring buf;
-  irs::bytes_output out(buf);
-  irs::encode::bitpack::write_block_simd(out, &source[0], source.size(), &enc_dec_buf[0]);
-
-  // read block
-  bytes_input in(buf);
-  std::vector<uint32_t> read(source.size());
-  irs::encode::bitpack::read_block_simd(in, source.size(), &enc_dec_buf[0], read.data());
+template<size_t N>
+void read_write_block_optimized(const std::array<uint32_t, N>& source) {
+  static_assert(128 == N);
+  std::array<uint32_t, N> enc_dec_buf;
+  std::fill_n(enc_dec_buf.data(), N, std::numeric_limits<uint32_t>::max());
+
+  auto pack_block = [](
+      const uint32_t* RESTRICT decoded,
+      uint32_t* RESTRICT encoded,
+      const uint32_t bits) noexcept {
+    std::memset(encoded, 0, sizeof(uint32_t) * N); // FIXME do we need memset???
```

Comment:
I vote for removing memset as `simdpackwithoutmask` does not requires output buffer to be zero filled or we should mark it so